dashboard/api_test.py

```diff
@@ -0,0 +1,402 @@
+                         if not name.startswith('_') and isinstance(value, str)]
+        assert sorted(all_constants) == sorted(api.CourseStatus.all_statuses())
+
+    def test_course_run_status(self):
+        """test for CourseRunStatus"""
+        for attr in ('NOT_ENROLLED', 'GRADE', 'READ_CERT', 'WILL_ATTEND', 'UPGRADE', 'NOT_PASSED'):
+            assert hasattr(api.CourseRunStatus, attr)
+
+    def test_corse_run_user_status(self):
+        """test for CourseRunUserStatus"""
+        ustat = api.CourseRunUserStatus(
+            status='status',
+            course_run='run',
+            enrollment_for_course='enroll'
+        )
+        assert ustat.status == 'status'
+        assert ustat.course_run == 'run'
+        assert ustat.enrollment_for_course == 'enroll'
+
+    def test_course_format_conditional_fields_struct(self):
+        """
+        test for CourseFormatConditionalFields:
+        checking the association has the right structure and key/value pairs
+        """
+        assert isinstance(api.CourseFormatConditionalFields.ASSOCIATED_FIELDS, dict)
+        for key in api.CourseFormatConditionalFields.ASSOCIATED_FIELDS:
+            assert key in api.CourseStatus.all_statuses()
+            assert isinstance(api.CourseFormatConditionalFields.ASSOCIATED_FIELDS[key], list)
+            for assoc in api.CourseFormatConditionalFields.ASSOCIATED_FIELDS[key]:
+                assert isinstance(assoc, dict)
+                assert 'course_run_field' in assoc
+                assert 'format_field' in assoc
+                assert assoc['course_run_field'] not in ['', None]
+                assert assoc['format_field'] not in ['', None]
+
+    def test_course_format_conditional_fields_get(self):
+        """test for CourseFormatConditionalFields.get_assoc_field"""
+        assert api.CourseFormatConditionalFields.get_assoc_field('foobar') == []
+        assert len(api.CourseFormatConditionalFields.get_assoc_field(api.CourseStatus.OFFERED)) == 2
+
+
+class CourseMixin(TestCase):
+    """Base class for APIs tests"""
+
+    @classmethod
+    def setUpClass(cls):
+        super(CourseMixin, cls).setUpClass()
+        cls.course = CourseFactory.create(title="Title")
+
+    def create_run(self, course=None, start=None, end=None, enr_start=None, enr_end=None, edx_key=None):
+        """helper function to create course runs"""
+        # pylint: disable=too-many-arguments
+        run = CourseRunFactory.create(
+            course=course or self.course,
+            title="Title",
+            start_date=start,
+            end_date=end,
+            enrollment_start=enr_start,
+            enrollment_end=enr_end,
+        )
+        if edx_key is not None:
+            run.edx_course_key = edx_key
+            run.save()
+        return run
+
+
+class FormatRunTest(CourseMixin):
+    """Tests for the format_course_for_dashboard function"""
+
+    def test_format_run_no_run(self):
```

Comment:
Can you add a test that uses one of the statuses in `CourseFormatConditionalFields` and verifies that it added the fields correctly?